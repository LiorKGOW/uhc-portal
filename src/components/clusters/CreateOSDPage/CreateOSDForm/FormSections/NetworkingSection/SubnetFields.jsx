import React from 'react';
import PropTypes from 'prop-types';
import {
  GridItem,
} from '@patternfly/react-core';
import { Field } from 'redux-form';
import ReduxVerticalFormGroup from '../../../../../common/ReduxFormComponents/ReduxVerticalFormGroup';
import AvailabilityZoneSelection from './AvailabilityZoneSelection';
import { required, validateUniqueAZ } from '../../../../../../common/validators';

const SingleSubnetFieldsRow = ({
  showLabels = false,
  index,
  selectedRegion,
  isMultiAz,
}) => {
  const azValidations = [
    required,
    isMultiAz && validateUniqueAZ,
  ].filter(Boolean);

  return (
    <>
      <GridItem className="vpc-input-field" span={3}>
        <Field
          component={AvailabilityZoneSelection}
          name={`az_${index}`}
          label={showLabels ? 'Availability zone' : null}
          validate={azValidations}
          region={selectedRegion}
        />
      </GridItem>
      <GridItem span={3}>
        <Field
          component={ReduxVerticalFormGroup}
          name={`private_subnet_id_${index}`}
          label={showLabels ? 'Private subnet ID' : null}
          type="text"
          validate={required}
        />
      </GridItem>
      <GridItem span={3}>
        <Field
          component={ReduxVerticalFormGroup}
          name={`public_subnet_id_${index}`}
          label={showLabels ? 'Public subnet ID' : null}
          type="text"
          validate={required}
        />
      </GridItem>
      <GridItem span={3} />
    </>
  );
};

const SubnetFields = ({
  selectedRegion,
  isMultiAz,
}) => (
  <>
    <SingleSubnetFieldsRow
      showLabels
      index={0}
      selectedRegion={selectedRegion}
      isMultiAz={isMultiAz}
    />
    {
    isMultiAz && (
      <>
        <SingleSubnetFieldsRow
          index={1}
          selectedRegion={selectedRegion}
          isMultiAz={isMultiAz}
        />
        <SingleSubnetFieldsRow
          index={2}
          selectedRegion={selectedRegion}
          isMultiAz={isMultiAz}
        />
      </>
    )
  }
  </>
);

SingleSubnetFieldsRow.propTypes = {
  selectedRegion: PropTypes.string,
  index: PropTypes.number,
  showLabels: PropTypes.bool,
  isMultiAz: PropTypes.bool,
};

SubnetFields.propTypes = {
  selectedRegion: PropTypes.string,
  isMultiAz: PropTypes.bool,
};

export { SingleSubnetFieldsRow };
export default SubnetFields;
