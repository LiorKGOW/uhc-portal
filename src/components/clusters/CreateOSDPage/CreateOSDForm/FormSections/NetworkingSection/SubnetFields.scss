.vpc-input-field .pf-v5-c-input-group.valid-field:only-child {
  margin-bottom: var(--pf-v5-global--spacer--lg);
}
