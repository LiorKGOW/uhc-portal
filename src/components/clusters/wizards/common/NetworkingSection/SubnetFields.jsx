import React from 'react';
import PropTypes from 'prop-types';
import { GridItem } from '@patternfly/react-core';
import { Field } from 'redux-form';
import ReduxVerticalFormGroup from '~/components/common/ReduxFormComponents/ReduxVerticalFormGroup';
import { required, validateUniqueAZ, validateValueNotPlaceholder } from '~/common/validators';
import AvailabilityZoneSelection, { PLACEHOLDER_VALUE } from './AvailabilityZoneSelection';

import './SubnetFields.scss';

const SingleSubnetFieldsRow = ({
  showLabels = false,
  index,
  selectedRegion,
  isMultiAz,
  privateLinkSelected,
}) => {
  const azValidations = [
    isMultiAz && validateUniqueAZ,
    validateValueNotPlaceholder(PLACEHOLDER_VALUE),
    required,
  ].filter(Boolean);

  return (
    <>
      <GridItem className="vpc-input-field" md={3}>
        <Field
          component={AvailabilityZoneSelection}
          name={`az_${index}`}
          label={showLabels ? 'Availability zone' : null}
          validate={azValidations}
          region={selectedRegion}
        />
      </GridItem>
      <GridItem md={3}>
        <Field
          component={ReduxVerticalFormGroup}
          name={`private_subnet_id_${index}`}
          label={showLabels ? 'Private subnet ID' : null}
          type="text"
          isRequired
          validate={required}
        />
      </GridItem>
      {!privateLinkSelected && (
        <GridItem md={3}>
          <Field
            component={ReduxVerticalFormGroup}
            name={`public_subnet_id_${index}`}
            label={showLabels ? 'Public subnet ID' : null}
            type="text"
            isRequired
            validate={required}
          />
        </GridItem>
      )}
      <GridItem md={privateLinkSelected ? 6 : 3} />
    </>
  );
};

const SubnetFields = ({ selectedRegion, isMultiAz, privateLinkSelected }) => (
  <>
    <SingleSubnetFieldsRow
      showLabels
      index={0}
      selectedRegion={selectedRegion}
      isMultiAz={isMultiAz}
      privateLinkSelected={privateLinkSelected}
    />
    {isMultiAz && (
      <>
        <SingleSubnetFieldsRow
          index={1}
          selectedRegion={selectedRegion}
          isMultiAz={isMultiAz}
          privateLinkSelected={privateLinkSelected}
        />
        <SingleSubnetFieldsRow
          index={2}
          selectedRegion={selectedRegion}
          isMultiAz={isMultiAz}
          privateLinkSelected={privateLinkSelected}
        />
      </>
    )}
  </>
);

SingleSubnetFieldsRow.propTypes = {
  selectedRegion: PropTypes.string,
  index: PropTypes.number,
  showLabels: PropTypes.bool,
  isMultiAz: PropTypes.bool,
  privateLinkSelected: PropTypes.bool,
};

SubnetFields.propTypes = {
  selectedRegion: PropTypes.string,
  isMultiAz: PropTypes.bool,
  privateLinkSelected: PropTypes.bool,
};

export { SingleSubnetFieldsRow };
export default SubnetFields;
