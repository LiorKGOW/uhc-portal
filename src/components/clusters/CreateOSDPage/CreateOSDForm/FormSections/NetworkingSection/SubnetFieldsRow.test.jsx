import React from 'react';
import { shallow } from 'enzyme';
import SubnetFields from './SubnetFields';

describe('<SubnetFields />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SubnetFields selectedRegion="fake-region" />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
