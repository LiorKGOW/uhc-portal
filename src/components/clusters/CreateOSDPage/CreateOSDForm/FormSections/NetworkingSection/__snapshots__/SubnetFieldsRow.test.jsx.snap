// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubnetFields /> should render 1`] = `
<Fragment>
  <SingleSubnetFieldsRow
    index={0}
    selectedRegion="fake-region"
    showLabels={true}
  />
</Fragment>
`;
